src/DatasourceEntriesApi.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Storyblok-Api.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SensioLabs\Storyblok\Api;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use SensioLabs\Storyblok\Api\Domain\Value\Dto\Pagination;
+use SensioLabs\Storyblok\Api\Domain\Value\Total;
+use SensioLabs\Storyblok\Api\Response\DatasourceEntriesResponse;
+use Webmozart\Assert\Assert;
+
+final readonly class DatasourceEntriesApi implements DatasourceEntriesApiInterface
+{
+    public function __construct(
+        private StoryblokClientInterface $client,
+        private LoggerInterface $logger = new NullLogger(),
+    ) {
+    }
+
+    public function all(?Pagination $pagination = null): DatasourceEntriesResponse
+    {
+        return $this->getDatasourceEntries([], $pagination);
+    }
+
+    public function allByDatasource(string $datasource, ?Pagination $pagination = null): DatasourceEntriesResponse
+    {
+        Assert::regex($datasource, '/^[a-z0-9-]+$/');
+
+        return $this->getDatasourceEntries(['datasource' => $datasource], $pagination);
+    }
+
+    public function allByDimension(string $dimension, ?Pagination $pagination = null): DatasourceEntriesResponse
+    {
+        Assert::regex($dimension, '/^[a-z0-9-]+$/');
+
+        return $this->getDatasourceEntries(['datasource' => $dimension], $pagination);
+    }
+
+    public function allByDatasourceDimension(string $datasource, string $dimension, ?Pagination $pagination = null): DatasourceEntriesResponse
```

Comment:
is this needed at all? maybe we shouldnt allow this here. i think this belongs more to the Datasource Api not the entries.

Reply:
It returns a list of entries, i feel it's very similar to byContentType for instance in the StoriesApi